#28 지역 삭제

0hun opened the pull request.

0hun committed:
#28 지역 삭제

지역 삭제 메소드 추가

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/RegionController.java:
```diff
@@ -38,4 +49,48 @@
+   * @return ResponseEntity
+   */
+  @LoginCheck(userLevel = UserLevel.ADMIN)
+  @PostMapping("/name")
```
이거 밑의 PR에서 본 내용 같은데 타겟 브랜치가 잘못설정된걸까요?

0hun replied:
브렌치를 잘못설정하여 다시 설정하였습니다.

0hun committed:
Merge branch 'develop' into issue/28

0hun committed:
Merge branch 'issue/27' into issue/28

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/RegionController.java:
```diff
@@ -69,4 +74,25 @@
+   * @return ResponseEntity
+   */
+  @LoginCheck(userLevel = UserLevel.ADMIN)
+  @DeleteMapping("/name")
```
여기도 name이 붙어있네요~

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/RegionController.java:
```diff
@@ -69,4 +74,25 @@
+    boolean isExistsRegion =  regionService.existsByName(name);
+
+    if (isExistsRegion) {
+      return RESPONSE_BAD_REQUEST;
```
왜 요청이 실패했는지도 알려줄 수 있는 코드면 좋겠습니다~

0hun committed:
#28 리뷰 반영

name만 추가하는게 아니라 지역을 추가하는것이므로 name url을 붙이지 않아도
되므로 수정

삭제시 존재하지 않는 지역을 삭제하려고 했을때, message return

0hun committed:
Merge branch 'issue/28' of https://github.com/f-lab-edu/hi-bus-go into issue/28

0hun committed:
#28 지역 삭제 메소드 수정

지역 삭제시 아이디 값을 받아서 삭제하도록 수정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/RegionController.java:
```diff
@@ -76,21 +77,20 @@
   @DeleteMapping()
-  public ResponseEntity<?> deleteRegion(@RequestBody @Valid RegionDto regionDto) {
-    String name = regionDto.getName();
+  public ResponseEntity<?> deleteRegion(@RequestParam int id) {
```
이렇게 된다면 path variable을 사용하는게 낫지 않을까요?

0hun replied:
넵 수정하겠습니다.

0hun committed:
#28 리뷰반영

id 값을 path variable로 받도록 수정

0hun committed:
#28 지역 삭제

지역이 존재 하지 않았을 경우 체크하는 것이므로 not(!)을 붙여 체크하도록 수정
existsById 쿼리 추가

f-lab-dev commented on src/main/resources/mapper/region.xml:
```diff
@@ -24,6 +24,13 @@
     </select>
 
+    <select id="existsById" resultType="boolean" parameterType="Integer">
+        SELECT COUNT(*)
```
~~존재하는지 여부 리턴이 목적이라면 boolean으로 리턴할 수 있지 않을까요?~~
잘못봤습니다~ 맞게 되어있네요
